Add direction (vision, strategy, plans) template page (#4599)
sourcegraph/about: handbook/product/product_management/direction_template.md

# Vision, Strategy, and Roadmap Template

> > ℹ️  The following is a template for use on the pages of specific product & engineering teams, used to record and share the vision, strategy, and roadmap.<br>Copy to your team's page and complete accordingly. Other sources of truth (such as Q3 one-pagers) should have their content integrated here and then be marked as deprecated. Be sure to also replace instances of `ProductArea` with your actual product area name.

Reviewer: ```suggestion
> ℹ️  The following is a template for use on the pages of specific product & engineering teams, used to record and share the vision, strategy, and roadmap.<br>Copy to your team's page and complete accordingly. Other sources of truth (such as Q3 one-pagers) should have their content integrated here and then be marked as deprecated. Be sure to also replace instances of `ProductArea` with your actual product area name. Unless otherwise specified, each section should be a paragraph or two but you're welcome to use your own judgement if you think a deeper dive is appropriate.
```

> > As you write this document, keep in mind that focusing on problems to solve more than lists of features tends to be a better way to approach communicating with customers, as it leaves room for learning between what you know now and what you deliver in the end. This is not to say you never talk about features, but especially as you get [further out in planning](https://medium.com/@jobv/the-nearsighted-roadmap-57fa57b5906a) the blend of features vs. problems to solve should move more towards the latter.

Reviewer: ```suggestion
> As you write this document, keep in mind that focusing on problems to solve more than lists of features tends to be a better way to approach communicating with people interested in your product area (customers, sales, analysts, etc.) as it leaves room for learning between what you know now and what you deliver in the end. This is not to say you never talk about features, but especially as you get [further out in planning](https://medium.com/@jobv/the-nearsighted-roadmap-57fa57b5906a) the blend of features vs. problems to solve should move more towards the latter.
```

> Be sure also to explain [why something is part of your direction](https://medium.com/leadership-motivation-and-impact/the-power-of-starting-with-why-f8e491392ef8). What you are doing is interesting, but why you have chosen that of all the possibilities is often even more enlightening.

---

# MyProductArea Direction

This page outlines the vision, strategy and goals of the MyProductArea team. You can use this section to describe what your scope is, and where your product is today.

Quicklinks:

> - [MyProductArea Backlog](https://about.sourcegraph.com/) <!-- Update links with your page -->

Reviewer: ```suggestion
- [MyProductArea Roadmap](https://sourcegraph.productboard.com/roadmap/) <!-- Update links with your page -->
```

Describe the target by which we prioritize:

- Market or audience
- Personas
- Jobs or Use Cases
- Industries
- What matters to us when we prioritize products and features
- Contextual things that matter to Sourcegraph or the product

# Where we are now

- Where is the team's area of ownership in terms of its maturity? Is it new and basic, or complete and lovable? Are different features at different levels?
- What did we achieve in the last few months?
- What key learnings did we recently make?
- What is on the critical path for growth?
- How does the product fit within Sourcegraph as a whole?

## Competitive landscape

- Who is the competition for this product area?
- What recent developments in competition is impacting sales, marketing and product strategy?

## Analyst landscape

- Are there analysts tracking this product area?
- How are analysts positioning the product? What are areas of improvement?

# What's next and why

What are we going to do in the next quarter, and how does it accomplish company goals and the product vision?

> 

Author: How about adding a subsection here that is specifically the Quarterly plans? 

That way we keep the longer term, loosely defined what's next under Themes, but we can also answer what the sales and CE team will be hoping to see here: what's coming up in the next quarter?

> ## Themes

Reviewer: @malomarrec I see what you mean with this three year outlook not fitting within the quarterly 'what's next' bit. Maybe lets move them to the top level after guiding principles?

Author: That works!

- What is on the critical path for growth?
- How does the product fit within Sourcegraph as a whole?

## Competitive landscape

> 

Reviewer: Good call @Joelkw, I'll add something up at the top. Most sections should be a paragraph or two, but it is up to your final judgement.

- Who is the competition for this product area?
- What recent developments in competition is impacting sales, marketing and product strategy?

## Analyst landscape

> 

Reviewer: Good idea @Joelkw added

```suggestion

- If you are not currently involved with analyst discussions for your product area, you can reach out to Christine for context here and to stay in the loop in the future.
```

> 

Reviewer 2: Sweet!

# Vision, Strategy, and Roadmap Template

> > ℹ️  The following is a template for use on the pages of specific product & engineering teams, used to record and share the vision, strategy, and roadmap.<br>Copy to your team's page and complete accordingly.

Reviewer: ```suggestion
> ℹ️  The following is a template for use on the pages of specific product & engineering teams, used to record and share the vision, strategy, and roadmap.<br>Copy to your team's page and complete accordingly. Other sources of truth (such as Q3 one-pagers) should have their content integrated here and marked as deprecated.
```

> As you write this document, keep in mind that focusing on problems to solve more than lists of features tends to be a better way to approach communicating with customers, as it leaves room for learning between what you know now and what you deliver in the end. This is not to say you never talk about features, but especially as you get [further out in planning](https://medium.com/@jobv/the-nearsighted-roadmap-57fa57b5906a) the blend of features vs. problems to solve should move more towards the latter.

> Be sure also to explain [why something is part of your direction](https://medium.com/leadership-motivation-and-impact/the-power-of-starting-with-why-f8e491392ef8). What you are doing is interesting, but why you have chosen that of all the possibilities is often even more enlightening.

---

# MyProductArea Direction

This page outlines the vision, strategy and goals of the MyProductArea team. You can use this section to describe what your scope is, and where your product is today.

Quicklinks:

> - [MyProductArea Roadmap](https://sourcegraph.productboard.com/roadmap/) <!-- Update links with your page -->

Reviewer: These were no longer updated in Q2, in Q3 we moved to one-pagers which will be integrated here.

```suggestion
```

# Vision, Strategy, and Roadmap Template

> ℹ️  The following is a template for use on the pages of specific product & engineering teams, used to record and share the vision, strategy, and roadmap.<br>Copy to your team's page and complete accordingly.

> As you write this document, keep in mind that focusing on problems to solve more than lists of features tends to be a better way to approach communicating with customers, as it leaves room for learning between what you know now and what you deliver in the end. This is not to say you never talk about features, but especially as you get [further out in planning](https://medium.com/@jobv/the-nearsighted-roadmap-57fa57b5906a) the blend of features vs. problems to solve should move more towards the latter.

> > Be sure also to include why something is part of your direction. What you are doing is interesting, but why you have chosen that of all the possibilities is often even more enlightening.

Reviewer: ```suggestion
> Be sure also to explain [why something is part of your direction](https://medium.com/leadership-motivation-and-impact/the-power-of-starting-with-why-f8e491392ef8). What you are doing is interesting, but why you have chosen that of all the possibilities is often even more enlightening.
```

# Vision, Strategy, and Roadmap Template

> > ℹ️  The following is a template for use on the pages of specific product & engineering teams, used to record and share the vision, strategy, and roadmap.<br>Copy to your team's page and complete accordingly. Other sources of truth (such as Q3 one-pagers) should have their content integrated here and then be marked as deprecated. Be sure to also replace instances of `ProductArea` with your actual product area name. Unless otherwise specified, each section should be a paragraph or two but you're welcome to use your own judgement if you think a deeper dive is appropriate.

Reviewer: ```suggestion
> ℹ️  The following is a template for use on the pages of specific product & engineering teams, used to record and share the vision, strategy, and roadmap.<br>Copy to your team's page and complete accordingly. Other sources of truth (such as Q3 one-pagers, existing goals pages) should have their content integrated here and then be marked as deprecated. Be sure to also replace instances of `ProductArea` with your actual product area name. Unless otherwise specified, each section should be a paragraph or two but you're welcome to use your own judgement if you think a deeper dive is appropriate.
```

> Be sure also to explain [why something is part of your direction](https://medium.com/leadership-motivation-and-impact/the-power-of-starting-with-why-f8e491392ef8). What you are doing is interesting, but why you have chosen that of all the possibilities is often even more enlightening.

---

# ProductArea Direction

This page outlines the vision, strategy and goals of the ProductArea team. You can use this section to describe what your scope is, and where your product is today.

Quicklinks:

- [ProductArea Backlog](https://about.sourcegraph.com/) <!-- Update links with your page -->
- [Latest demo](https://about.sourcegraph.com/)
- [Documentation](https://about.sourcegraph.com/)

## Mission, Vision & Guiding Principles

### Mission

> What is the team's reason for existing? What is the fundamental value they aim to provide?

Reviewer: ```suggestion
What is the team's reason for existing? What is the fundamental value they aim to provide? Who are your key customers/stakeholders and why, and are there any customers/stakeholders that you're explicitly choosing not to address that people might have expected you to?
```

### Vision

#### 1 Year vision

> Add a sentence or two here indicating what the most important things you want to achieve/problems you want to solve in the next year are. These should in most cases move your three and ten year visions forward.

#### 3 Year vision

> Add a sentence or two here describing what your product will be like in three years, also focusing on the problems you'll solve and the value you'll deliver. This should be grounded in reality, but three years is a lot of time - be ambitious!

#### 10 year vision

> A sentence or two on your ten year vision is where you can really go wild and describe the problems you want to solve and solutions you want to deliver unbound by limitations of technology or scope. Be inspiring and look to the our [10 and 30 year strategies](../../company/strategy.md#sourcegraph-strategy) for ideas.

### Guiding principles

Identify what key principles, insights, and decisions are important to the strategy or a set of key hypotheses and set of key questions to validate or invalidate them.

Describe the target by which we prioritize:

- Market or audience

> - Personas

Reviewer: FYI @abschu

```suggestion
- [Personas](../../marketing/personas.md)
```

> # <Product> direction

Reviewer 3: ```suggestion
# Vision, Strategy, and Roadmap Template

> ℹ️  The following is a template for use on the pages of specific product & engineering teams, used to record and share the vision, strategy, and roadmap.<br>Copy to your team's page and complete accordingly.

<hr>

# <Product> direction
```

Reviewer 3: Think it's useful to make it super clear that this is a template - it should be obvious, but just in case 😁

Author: Was outdated by one of muy commits, but incorporated it and pushing now.

This page outlines the vision, strategy and goals of the <Product> team.

# Vision

#### 1 Year vision

> Some vision

#### 3 Year vision

> Some vision

#### 10 year vision

> Some vision

> ## Mission

Reviewer: Maybe move mission above vision since it provides context for it?

> TODO

Reviewer 3: ```suggestion
What is the team's reason for existing? What is the fundamental value they aim to provide?
```

> 

Reviewer 3: ```suggestion

# Guiding principles

Identify what key principles, insights, and decisions are important to the strategy or a set of key hypotheses and set of key questions to validate or invalidate them.

Describe the target by which we prioritize:
- Market or audience
- Personas 
- Jobs or Use Cases
- Industries
- What matters to us when we prioritize products and features
- Contextual things that matter to Sourcegraph or the product

```

Reviewer 3: I'm a MASSIVE fan of the guiding principles to steer the team.

Author: @dan-mckean where does this belong in your opinion? Is it in strategy? Or rather right under mission?

Author: Reordered to match your suggestion. Sounds like the Guiding Principles are more relevant in the strategy though, as they are interdependent with the Themes

Reviewer: If guiding principles are indeed relevant to both themes and vision, then I agree the right place for them is out of the three year strategy section and next to mission.

Reviewer: Another idea @malomarrec, maybe there just doesn't need to be a "3 year strategy and themes" heading?

- Strategy could be combined with Mission into "Mission & Strategy" up at the top
- "Guiding Principles" could be added just below "Mission & Strategy" at the top level
- "Themes" could be moved inside of "What's next & why"

I made a PR https://github.com/sourcegraph/about/pull/4600/files with the changes to consider.

Reviewer: Resolving since I merged in that PR.

> # Where we are now

Reviewer: I would propose removing this as a separate heading, and making it part of the introduction for each direction page.

> - Where is the product like in maturity?

Reviewer: ```suggestion
- Where is the product area in terms of its maturity? Is it new and basic, or complete and lovable? Are different features at different levels?
```

> - Where is the product like in maturity?

Reviewer 3: ```suggestion
- Where is the team's area of ownership like in maturity?
```

Reviewer 3: Not always a product.

Author: :+1: thanks! Added in latest commit.

- What did we achieve in the last few months?
- What key learnings did we recently make?
- What is on the critical path for growth?
- How does the product fit within Sourcegrah as a whole?

# Competitive landscape

- Who is the competition for this product area?
- What recent developments in competition is impacting sales, marketing and product strategy?

# Analyst landscape

- Are there analysts tracking this product area?
- How are analysts positioning the product? What are areas of improvement?

# 3 year strategy and themes

## Strategy

> What is wer strategy defined in a few steps?

Reviewer: ```suggestion
What is your strategy defined in a few steps?
```

# MyProductArea Direction

> This page outlines the vision, strategy and goals of the MyProductArea team. You can use this section to describe what your scope is, and where your product is today.

Reviewer: What do you think about the following? I'm not sure if this philosophy is controversial if not, but if so I could propose it in a separate PR.

```suggestion
This page outlines the vision, strategy and goals of the MyProductArea team. You can use this section to describe what your scope is, and where your product is today.

As you write this document, keep in mind that focusing on problems to solve more than lists of features tends to be a better way to approach communicating with customers, as it leaves room for learning between what you know now and what you deliver in the end. This is not to say you never talk about features, but especially as you get [further out in planning](https://medium.com/@jobv/the-nearsighted-roadmap-57fa57b5906a) the blend of features vs. problems to solve should move more towards the latter.
```

Author: I don't think it is controversial, Christina repeats this over and over again to explain how our team works to other teams. Adding.

# Vision, Strategy, and Roadmap Template

> ℹ️  The following is a template for use on the pages of specific product & engineering teams, used to record and share the vision, strategy, and roadmap.<br>Copy to your team's page and complete accordingly.

> <hr>

Reviewer: Not a huge deal but maybe better to do this in native markdown

```suggestion
---
```

## Mission

What is the team's reason for existing? What is the fundamental value they aim to provide?

# Where we are now

- Where is the team's area of ownership in terms of its maturity? Is it new and basic, or complete and lovable? Are different features at different levels?
- What did we achieve in the last few months?
- What key learnings did we recently make?
- What is on the critical path for growth?
- How does the product fit within Sourcegraph as a whole?

## Competitive landscape

- Who is the competition for this product area?
- What recent developments in competition is impacting sales, marketing and product strategy?

## Analyst landscape

- Are there analysts tracking this product area?
- How are analysts positioning the product? What are areas of improvement?

# 3 year strategy and themes

## Strategy

What is your strategy defined in a few steps?

## Themes

> What are high level themes or principles we use to guide prioritisation? What are we focusing on in the next few years?

Reviewer: I am trying to differentiate themes as part of the three year strategy from the vision section above. What do you think?

```suggestion
What are high level themes or principles you are using to guide prioritization within your three year strategy? What are the groupings of features/problems to solve that you are focusing on in that period?
```
